### PR TITLE
remove padding-top for TextField

### PR DESCRIPTION
**Pull request by Ksj14-kumar**
fixed: remove padding-top for TextField if no label is provided.

### CODE REVIEW

**Comment by vczb** [comment]
I will check and back to you ASAP. I can also help to figure out why the CI is broking

File: src/packages/TextField/styles.ts [hunk 1]
```diff
@@ -14,7 +15,7 @@ export const TextField = styled.input`
     outline: none;
     height: 100%;
     letter-spacing: 0.1rem;
-    padding: 1.5rem 2.2rem 0 1.5rem;
+    padding: ${label ? '1.5rem 2.2rem 0 1.5rem' : '0 2.2rem 0 1.5rem'};
```
Comment (vczb) [inline src/packages/TextField/styles.ts:18]: nice :+1:

File: src/packages/TextField/styles.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import styled, { css } from 'styled-components'
+import { CustomInputHTMLAttributes } from './index'
 
-export const TextField = styled.input`
-  ${({ theme }) => css`
+export const TextField = styled.input<CustomInputHTMLAttributes>`
```
Comment (vczb) [inline src/packages/TextField/styles.ts:4]: lets import label from TextFieldProps

```
export const TextField = styled.input<Pick<TextFieldProps, 'label'>>`
...
```

**Review by vczb — COMMENTED** [review]
you can fix the build errors with the command `yarn prettier:format`

follow here some pseudo code suggesting improvements

src/packages/TextField/index.tsx
```
import React from 'react'
import { InputHTMLAttributes } from 'react'
import Label from '../Label'
import * as S from './styles'

export type TextFieldProps = {
  label?: string
} & InputHTMLAttributes<HTMLInputElement>

const TextField = ({ label, ...props }: TextFieldProps) => (
  <Label title={label}>
    <S.TextField {...props} />
  </Label>
)

export default TextField
```

src/packages/Label/index.tsx

```
import React from 'react'
import { LabelHTMLAttributes } from 'react'
import * as S from './styles'

export type LabelProps = {
  title?: string
  children?: React.ReactNode
} & LabelHTMLAttributes<HTMLLabelElement>

const Label = ({ title, children, ...props }: LabelProps) => (
  <S.Label {...props}>
    {title && <S.Title>{title}</S.Title>}
    {children}
  </S.Label>
)

export default Label
```

src/packages/TextField/styles.ts
```
import styled, { css } from 'styled-components'
import { TextFieldProps } from '.'

export const TextField = styled.input<Pick<TextFieldProps, 'label'>>`
  ${({ theme, label }) => css`
    background: ${theme.colors.base.white};
    border-radius: ${theme.border.xxsmall};
    border: 0.2rem solid ${theme.colors.primary.lighter};
    border-radius: ${theme.border.xxsmall};
    color: ${theme.colors.primary.dark};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.weights.regular};
    font-family: ${theme.font.family.primary};
    transition: ${theme.transitions.default};
    outline: none;
    height: 100%;
    letter-spacing: 0.1rem;
    padding: ${label ? '1.5rem 2.2rem 0 1.5rem' : '0 2.2rem 0 1.5rem'};
    &::placeholder {
      color: ${theme.colors.primary.medium};
    }
    &:hover {
      border: 0.2rem solid ${theme.colors.secondary.lighter};
    }
    &:focus {
      border: 0.2rem solid ${theme.colors.secondary.light};
    }
  `}
`
```

Let me know if you have a different idea

Thanks for your PR

File: yarn.lock
[the remark this answers is not in the document]
Comment (Ksj14-kumar) [reply yarn.lock:1]: okay, I fix this.

**Comment by Ksj14-kumar** [comment]
> you can fix the build errors with the command `yarn prettier:format`
> 
> follow here some pseudo code suggesting improvements
> 
> src/packages/TextField/index.tsx
> 
> ```
> import React from 'react'
> import { InputHTMLAttributes } from 'react'
> import Label from '../Label'
> import * as S from './styles'
> 
> export type TextFieldProps = {
>   label?: string
> } & InputHTMLAttributes<HTMLInputElement>
> 
> const TextField = ({ label, ...props }: TextFieldProps) => (
>   <Label title={label}>
>     <S.TextField {...props} />
>   </Label>
> )
> 
> export default TextField
> ```
> 
> src/packages/Label/index.tsx
> 
> ```
> import React from 'react'
> import { LabelHTMLAttributes } from 'react'
> import * as S from './styles'
> 
> export type LabelProps = {
>   title?: string
>   children?: React.ReactNode
> } & LabelHTMLAttributes<HTMLLabelElement>
> 
> const Label = ({ title, children, ...props }: LabelProps) => (
>   <S.Label {...props}>
>     {title && <S.Title>{title}</S.Title>}
>     {children}
>   </S.Label>
> )
> 
> export default Label
> ```
> 
> src/packages/TextField/styles.ts
> 
> ```
> import styled, { css } from 'styled-components'
> import { TextFieldProps } from '.'
> 
> export const TextField = styled.input<Pick<TextFieldProps, 'label'>>`
>   ${({ theme, label }) => css`
>     background: ${theme.colors.base.white};
>     border-radius: ${theme.border.xxsmall};
>     border: 0.2rem solid ${theme.colors.primary.lighter};
>     border-radius: ${theme.border.xxsmall};
>     color: ${theme.colors.primary.dark};
>     font-size: ${theme.font.sizes.small};
>     font-weight: ${theme.font.weights.regular};
>     font-family: ${theme.font.family.primary};
>     transition: ${theme.transitions.default};
>     outline: none;
>     height: 100%;
>     letter-spacing: 0.1rem;
>     padding: ${label ? '1.5rem 2.2rem 0 1.5rem' : '0 2.2rem 0 1.5rem'};
>     &::placeholder {
>       color: ${theme.colors.primary.medium};
>     }
>     &:hover {
>       border: 0.2rem solid ${theme.colors.secondary.lighter};
>     }
>     &:focus {
>       border: 0.2rem solid ${theme.colors.secondary.light};
>     }
>   `}
> `
> ```
> 
> Let me know if you have a different idea
> 
> Thanks for your PR

File: src/packages/TextField/styles.ts (hunk 2 above)
Comment (Ksj14-kumar) [reply src/packages/TextField/styles.ts:4]: @vczb , actually with this code as you provided above it is not work. and it looks weird.
sagu

Comment (Ksj14-kumar) [reply src/packages/TextField/styles.ts:4]: and also I set title: string | undefined inside the Label component,  
                                                                                                     
export type LabelProps = {
  title: string | undefined 
  children?: React.ReactNode
} & LabelHTMLAttributes<HTMLLabelElement>

const Label = ({ title, children, ...props }: LabelProps) => (
  <S.Label {...props}>
    {title && <S.Title>{title}</S.Title>}
    {children}
  </S.Label>
)

**Comment by Ksj14-kumar** [comment]
@vczb so tell me what can I do? I have sent a pull request.

**Comment by vczb** [comment]
hey @Ksj14-kumar you did great. 

folow here what changes you need to do:

## I did a short video 

https://user-images.githubusercontent.com/51221635/234939772-7edb047f-1aa5-405b-ab96-669d8602ac0f.mp4

**1. src/packages/Label/index.tsx**
image

**2. src/packages/TextField/index.tsx**
image


---

## PLEASE 
Squash all commits in a single one
image

let me know if you want some help to do this

---

Thanks again for your PR! you are always welcome :rocket:

**Comment by Ksj14-kumar** [comment]
@vczb Thanks

**Comment by Ksj14-kumar** [comment]
@vczb check this, I have  sent an updated pull request.

**Comment by vczb** [comment]
Hey @Ksj14-kumar great job, your PR looks good! :rocket:

We only need to squash all commits on a single one. There are several ways to do this I made a video showing one of those.

https://user-images.githubusercontent.com/51221635/235243314-e8d60298-52a2-41ca-8b94-7b504ec8cc17.mp4

---

follow the steps :

1. `git log`
2. copy the hash of your first commit
3. `git reset hash-of-the-commit`
4. `git add .`
5. `git commit --amend --no-edit`
6. `git push origin issue-#71 -f`

---

Any questions feel free to ping me.

After squash your PR will be ready to merge :fire: 

---

If you want to squash using a different approach you can take a look [here](https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together)

**Comment by Ksj14-kumar** [comment]
@vczb Thank you for your explanation.

**Comment by Ksj14-kumar** [comment]
@vczb if there is an issue please let me know.

**Review by vczb — APPROVED** [review]
LGTM

**Pull request merged**